packages/api/endpoints/executions.js

```diff
@@ -79,6 +86,38 @@ async function del(req, res) {
   return res.send({ message: 'Record deleted' });
 }
 
+/**
+ * Get execution history for a single granule or multiple granules
+ *
+ * @param {Object} req - express request object
+ * @param {Object} res - express response object
+ * @returns {Promise<Object>} the promise of express response object
+ */
+async function history(req, res) {
+  const { granuleId } = req.params;
+  const payload = req.body;
+  const granuleIds = granuleId ? [granuleId] : await getGranuleIdsForPayload(payload);
+
+  const knex = await getKnexClient({ env: process.env });
+
+  const executionPgModel = new ExecutionPgModel();
+  const granulePgModel = new GranulePgModel();
+  const granulesExecutionsPgModel = new GranulesExecutionsPgModel();
+
+  const granuleCumulusIds = await granulePgModel
+    .getRecordsCumulusIds(knex, (builder) => builder.whereIn('granule_id', granuleIds));
+
+  const executionCumulusIds = await granulesExecutionsPgModel
+    .getExecutionCumulusIdsFromGranuleCumulusIds(knex, granuleCumulusIds);
+
+  const workflowNames = await executionPgModel
+    .getWorkflowNamesFromExecutionCumulusIds(knex, executionCumulusIds);
+
+  return res.send([...new Set(workflowNames)]);
+}
+
+router.post('/history', validateBulkGranulesRequest, history);
```

Comment:
Are these endpoints always returning a set of workflows for a single granule ID or a set of granule IDs? If so I'd have two change requests:

- Change the route from `/history` to `/workflows`, since that's the data we're really returning
- Move the definition of these routes to `endpoints/granules`, since the data is being returned based on granule ID as an input/path parameter. You could end up having this same route where execution ARN is a path parameter, but where it is a granule ID, I think this belongs with the other granule routes

Reply:
You also wouldn't need to have moved `validateBulkGranulesRequest` in that case

Reply:
So this is interesting. (Also, as a note, I just changed this to return the full executions)

@laurenfrederick had originally said that since we're returning executions/info about executions, this should be encapsulated in the /executions endpoint. 

Though I do agree that having granuleId as a path parameter rather than arn could case some confusion. We could maybe nix the GET request to avoid that? Or we could move it into granules, would just want Lauren's input on that.

Reply:
+1 to this should be a granule endpoint.

Reply:
Perhaps consider /executions/search?   We're not really getting an execution's history....   I won't hold up on this either, given the arch discussion as I don't have an alternate *better* suggestion at the moment.

Reply:
We discussed this in the arch meeting and I was convinced by the argument that what we're really returning here are execution records, thus this is an executions endpoint which is simply filtered by granule IDs.

Reply:
I do kind of like the idea of a `/search` endpoint but it's kind of specific to searching by granules, so I don't know if `/search-by-granules` is a bit wordy.  Because I don't think we could make the endpoint logic generic enough to handle any kind of search

Reply:
Updated this to be /search-by-granules. I think having "search" in there does make more sense. And making it specific to granules is also important as noted. Maybe a little wordy, but I don't think overly so.